Qorus login

rishabhmalik759 opened the pull request.

Commit:
Bootstraped project
- Installed required library
- Added tsconfig and jest config
- Updated Readme
- Initialized logger

Commit:
Added qorus login functionality with tests

Commit:
Added login and edited project structure

Commit:
updated readme name

Commit:
Added qorul login functionality along with:
- eslint
- updated build config
- removed logger
- added catchAll decorator

Commit:
Added logger decorator

Foxhoundn commented on src/QorusLogin.ts:
```diff
@@ -0,0 +1,85 @@
+  /**JWT token returned after authentication*/
+  usrToken?: string;
+  /**Qore Technologies endpoint to authenticate the user */
+  endpoint = 'https://hq.qoretechnologies.com:8092/api/latest/public/login';
```
This cannot be here - every user needs to instantiate this class with their own endpoint

Foxhoundn replied:
Create a constructor with `endpoint`

Foxhoundn commented on test/QorusLogin.test.ts:
```diff
@@ -0,0 +1,79 @@
+      let result;
+      try {
+        result = await qorus.login({
+          user: 'rmalik',
```
move this to `.env` file thats in gitignore pls, lets not push our credentials to open source projects :)

Foxhoundn commented on src/QorusLogin.ts:
```diff
@@ -0,0 +1,85 @@
+   */
+  @log({ level: 'verbose' })
+  logout() {
+    this.usrToken = undefined;
```
We should also logout from the server here using the API

Foxhoundn commented on src/QorusLogin.ts:
```diff
@@ -0,0 +1,85 @@
+   * A setter to configure the Authentication endpoint
+   */
+  @log({ level: 'verbose' })
+  config(endpoint: string) {
```
lets name this `setConfig` or `changeConfig` please

Foxhoundn replied:
or let's change it to a proper setter `set endpoint`

Foxhoundn reviewed (changes requested):
Please see what I found so far

rishabhmalik759 commented:
Meeting notes (26 Aug 2022): 


- check if in browser or node, if the user calls login and if the token exist in cookies just load the user
- test for the check
- window localstorage to store toke (qorus-auth-token)
- on logout remove the token from localstorage

Commit:
Updated Qorus Integration
- Updated tests
- Added .env configurations

Commit:
removed unwanted console log

Commit:
Added LocalStorageManger
- updated tests
- updated login to use localstorage first with token validation

Foxhoundn commented on src/utils/LocalStorageManager.ts:
```diff
@@ -0,0 +1,32 @@
+const detectEnvironment = () => {
```
Add types please and export it too

Foxhoundn commented on src/QorusIntegration.ts:
```diff
@@ -0,0 +1,128 @@
+  pass: string;
+}
+
+enum ApiPaths {
```
Maybe we could support multiple API versions, so instead of this being hardcorded to `api/latest` it should be something like `${API_VERSION}` where the default is `latest`

Commit:
updated linting and localstorage

Commit:
Updated QorusIntegration and local storage manager
- Api version modification added
- Converted local storage manager from class to function
- Added types

Commit:
Added Inheritance with Qorus Instance

rishabhmalik759 commented:
Implementation changed
